api/controllers/product.js

```diff
@@ -107,8 +108,16 @@ module.exports = {
 
             res.send(filters);
         } catch (e) {
-            res.status(500).json({ message: e.message });
-
+            res.status(statusCodes.InternalServerError).json({ message: e.message });
+        }
+    },
+    singleProduct: async (req, res, next) => {
+        const productId = req.query.id;
```

Comment:
Once you've changed the url to use url params rather than query here you can use:
```suggestion
        const productId = req.params.id;
```